File: src/Page/MyCoursesPage.java
package Page;

import System.FunctionType;
import System.SystemMessage;

public class MyCoursesPage extends Page{
	
	private int numberOfCourses;
	
	public MyCoursesPage(String content) {
		super(content);
		setType(PageType.MY_COURSES_PAGE);
		setName("My Courses Page");
	}
	
	@Override
	public SystemMessage runPage() {
		showContent();	
		String input;
		while(true) {
			input = takeInput();
			try {
				int selection = Integer.parseInt(input);
				if (selection > numberOfCourses || selection < 0) {
					System.out.println("Enter a valid number");
				}
				else {
					return new SystemMessage(FunctionType.SELECT_MY_COURSE, PageType.SELECTED_MY_COURSE_PAGE, selection);
				}
			}
			catch (NumberFormatException e) {
				if (input.equalsIgnoreCase("q")) {
					return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.MAIN_MENU_PAGE, null);
				}
				else {
					System.out.println("Wrong Input");
				}
			}
		}
	}

	public int getNumberOfCourses() {
		return numberOfCourses;
	}

	public void setNumberOfCourses(int numberOfCourses) {
		this.numberOfCourses = numberOfCourses;
	}
}

File: iter3/python_development/Page/MyCoursesPage.py
from iter3.python_development.Page.Page import Page
from iter3.python_development.Page.PageType import PageType
from iter3.python_development.System.FunctionType import FunctionType
from iter3.python_development.System.SystemMessage import SystemMessage

class MyCoursesPage(Page):

    numberOfCourses = None

    def __init__(self, content):
        super().__init__(content)
        self.setType("MY_COURSES_PAGE")
        self.setName("My Courses Page")


    def runPage(self):
        self.showContent()
        inputFromPage = None
        while(True):
            inputFromPage = self.takeInput()
            try:
                selection = int(inputFromPage)
                if (selection > self.numberOfCourses or selection < 0):
                    print("Enter a valid number")
                else:
                    return SystemMessage("SELECT_MY_COURSE", "SELECTED_MY_COURSE_PAGE", selection)
            except:
                if(inputFromPage.lower() == "q"):
                    return SystemMessage("CHANGE_PAGE", "MAIN_MENU_PAGE", None)
                else:
                    print("Wrong Input")


    def getNumberOfCourses(self):
        return self.numberOfCourses
    
    def setNumberOfCourses(self, numberOfCourses):
        self.numberOfCourses = numberOfCourses